src/arrayMethodReduce.js

```diff
@@ -5,7 +5,27 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let newInitialValue;
+    let startIndex;
+
+    if (initialValue === undefined) {
+      if (typeof this[0] === 'string') {
+        startIndex = 1;
+        newInitialValue = this[0];
+      } else {
+        startIndex = 0;
+        newInitialValue = 0;
+      }
```

Comment:
We really don't need type check here. It only depends on existence of `initialValue`.